### PR TITLE
Sanntis

### PR DESCRIPTION
**Pull request by Marie59**
Add new tool for the marine Omics pilot of Fair-Ease. This tool (sanntis created by @SantiagoSanchezF) takes as an input the output of interproscan. 
The aim here is to create a workflow using Interproscan then sanntis. 

However, Sanntis also needs a genbank to function as an input. So the full workflow would be for instance:
1.  Get a nucleotide fasta
-> with 1. use prodigal get a protein fasta -> from this protein fasta use Interproscan and retrieve a tsv
-> with 1. construct the Genbank 
2. Take the tsv and the genbank and use them as input of sanntis

From there I still need to test if the pipeline works but on my side the sanntis tool itself seems to be working fine. 

@bgruening and @yvanlebras Do you know of any tool that constructs Genbanks ?? If not @SantiagoSanchezF told me it should be easy to create a tool but it would for instance be only for this Prodigal example.

As you both know all this gene stuff is not my domain of expertise so, I hope I am clear enough ;)

What do you think of this ? Thanks !!

### CODE REVIEW

**Comment by bgruening** [comment]
bakta and prokka can create Genbank files. Those are genome annotation tools https://usegalaxy.eu/root?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/bakta/bakta/1.9.3+galaxy0 however, it is for annotation of bacterial genomes, MAGs and plasmids. Is that what you need?

Prodigal in Galaxy can already output genbank files afaik: https://usegalaxy.eu/root?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/prodigal/prodigal/2.6.3+galaxy0

File: tools/marine_omics/.shed.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+categories:
+    - Ecology
+owner: ecology
+remote_repository_url: https://github.com/galaxyecology/tools-ecology/tree/master/tools/marine_omics
+homepage_url: https://github.com/fair-ease/Marine-Omics-Galaxy
+long_description: |
+  CTool for identifying biosynthetic gene clusters (BGCs) in genomic & metagenomic data
```
Comment (bgruening) [inline tools/marine_omics/.shed.yml:7]: ctool?

File: tools/marine_omics/sanntis.xml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        <token name="@VERSION_SUFFIX@">0</token>
+    </macros>
+    <edam_topics>
+        <edam_topic>topic_3387</edam_topic>
+    </edam_topics>
+    <requirements>
+        <requirement type="package" version="0.9.3.5">sanntis</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        sanntis --ip-file "$input_interpro" --outfile "output_sanntis.gff" "$input_genbank"
```
Comment (bgruening) [inline tools/marine_omics/sanntis.xml:14]: single quotes ;)

File: tools/marine_omics/sanntis.xml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    ]]></command>
+    <inputs>
+        <param name="input_interpro" type="data" format="tsv" label="Input a .tsv file coming  from InterProScan" help="Before using this tool you need to retrieve the right data by using the InterProScan tool"/>
```
Comment (bgruening) [inline tools/marine_omics/sanntis.xml:17]:
```suggestion
        <param name="input_interpro" type="data" format="tabular" label="Input the TSV file from InterProScan" help="Before using this tool you need to retrieve the right data by using the InterProScan tool"/>
```

If you check the output in Galaxy Interproscan it is a `tabular` format.

File: tools/marine_omics/sanntis.xml [hunk 4]
```diff
@@ -0,0 +1,53 @@
+<tool id="sanntis_marine" name="Sanntis biosynthetic gene clusters" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="20.01" license="MIT">
+    <description>in genomic and metagenomic data</description>
+    <macros>
+        <token name="@TOOL_VERSION@">0.1.0</token>
```
Comment (bgruening) [inline tools/marine_omics/sanntis.xml:4]:
```suggestion
        <token name="@TOOL_VERSION@">0.9.3.5</token>
```

File: tools/marine_omics/sanntis.xml (hunk 2 above)
Comment (Marie59) [reply tools/marine_omics/sanntis.xml:14]: rooh I always forget

File: tools/marine_omics/sanntis.xml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        <token name="@VERSION_SUFFIX@">0</token>
+    </macros>
+    <edam_topics>
+        <edam_topic>topic_3387</edam_topic>
+    </edam_topics>
+    <requirements>
+        <requirement type="package" version="0.9.3.5">sanntis</requirement>
```
Comment (bgruening) [inline tools/marine_omics/sanntis.xml:11]:
```suggestion
        <requirement type="package" version="@TOOL_VERSION@">sanntis</requirement>
```

File: tools/marine_omics/sanntis.xml (hunk 3 above)
Comment (Marie59) [reply tools/marine_omics/sanntis.xml:17]: Oh thanks !

File: tools/marine_omics/sanntis.xml (hunk 2 above)
Comment (bgruening) [reply tools/marine_omics/sanntis.xml:14]:
```suggestion
        sanntis --ip-file '$input_interpro' --outfile 'output_sanntis.gff' '$input_genbank'
```

File: tools/marine_omics/.shed.yml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    - Ecology
+owner: ecology
+remote_repository_url: https://github.com/galaxyecology/tools-ecology/tree/master/tools/marine_omics
+homepage_url: https://github.com/fair-ease/Marine-Omics-Galaxy
+long_description: |
+  Tool for identifying biosynthetic gene clusters (BGCs) in genomic & metagenomic data
+type: unrestricted
+auto_tool_repositories:
+  name_template: "{{ tool_id }}"
+  description_template: "Wrapper for marine omics tool: {{ tool_name }}."
```
Comment (bgruening) [inline tools/marine_omics/.shed.yml:11]: should that not be sanntis? also above the link to the github?

**Comment by Marie59** [comment]
> bakta and prokka can create Genbank files. Those are genome annotation tools https://usegalaxy.eu/root?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/bakta/bakta/1.9.3+galaxy0 however, it is for annotation of bacterial genomes, MAGs and plasmids. Is that what you need?
> 
> Prodigal in Galaxy can already output genbank files afaik: https://usegalaxy.eu/root?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/prodigal/prodigal/2.6.3+galaxy0

@SantiagoSanchezF what do you think of that ? Is this enough ?

**Comment by yvanlebras** [comment]
yes, in the example of sanntis, they are taking prodigal output genbank file as far as I know.

File: tools/marine_omics/.shed.yml (hunk 6 above)
Comment (Marie59) [reply tools/marine_omics/.shed.yml:11]: oh yes I went to quickly on that

**Review by bgruening — APPROVED** [review]
I think the shed.yml needs some more love the rest is ok!

**Comment by Marie59** [comment]
> I think the shed.yml needs some more love the rest is ok!

On it ;)

**Comment by Marie59** [comment]
Waouh a tool that created and working in 2hours ! Me happy !! Thanks !!

**Comment by bgruening** [comment]
Get it into https://github.com/usegalaxy-eu/usegalaxy-eu-tools and its deployed tomorrow ;)

**Comment by Marie59** [comment]
oh in which section does it go ?... @yvanlebras would biodiversity exploration be okay ?

**Comment by yvanlebras** [comment]
hmmm I am not sure... It will be best in something related to "Annotation" with interproscan. So maybe choose "Annotation" AND add "ANNOTATION" tools section to ecology domain

**Comment by Marie59** [comment]
hmm okay will do

**Pull request merged**